src/Zigzag_persistence/include/gudhi/filtered_zigzag_persistence.h

```diff
@@ -0,0 +1,470 @@
+      if (birth > death) {
+        std::swap(birth, death);
+      }
+
+      if (death - birth > shortestInterval) {
+        diag.emplace_back(birth, death, bar.dim);
+      }
+    }
+
+    return diag;
+  }
+
+  /**
+   * @brief Computes the births of the current essential cycles.
+   *
+   * @param diag Reference to vector where to store the infinite bars.
+   */
+  void _retrieve_infinite_bars(std::vector<Filtration_value_interval>& diag) {
+    auto birth = [this](internal_key birthKey) {
+      auto itBirth =  // lower_bound(x) returns leftmost y s.t. x <= y
+          std::lower_bound(
+              filtrationValues_.begin(), filtrationValues_.end(),
+              std::pair<internal_key, filtration_value>(birthKey, std::numeric_limits<filtration_value>::infinity()),
+              [](std::pair<internal_key, filtration_value> p1, std::pair<internal_key, filtration_value> p2) {
+                return p1.first < p2.first;
+              });
+      if (itBirth == filtrationValues_.end() || itBirth->first > birthKey) {
+        --itBirth;
+      }
+      return itBirth->second;
+    };
+
+    auto stream_infinite_interval = [&](dimension_type dim, internal_key birthIndex) {
+      if (dimMax_ == -1 || (dimMax_ != -1 && dim < dimMax_))
+        diag.emplace_back(birth(birthIndex), Filtration_value_interval::inf, dim);
+    };
+
+    pers_.get_current_infinite_intervals(stream_infinite_interval);
+  }
+};  // end class Filtered_zigzag_persistence_with_storage
+
+/**
+ * @ingroup zigzag_persistence
+ *
+ * @brief Class computing the zigzag persistent homology of a zigzag filtration. Algorithm based on \cite zigzag.
+ * 
+ * @tparam FilteredZigzagOptions Structure following the @ref FilteredZigzagOptions concept.
+ * Default value: @ref Default_filtered_zigzag_options.
+ */
+template <class FilteredZigzagOptions = Default_filtered_zigzag_options>
+class Filtered_zigzag_persistence {
+ public:
+  using Options = FilteredZigzagOptions;                       /**< Zigzag options. */
+  using internal_key = typename Options::internal_key;         /**< Key and index type, has to be signed. */
+  using face_key = typename Options::face_key;                 /**< Face ID type from external inputs. */
+  using filtration_value = typename Options::filtration_value; /**< Type for filtration values. */
+  using dimension_type = typename Options::dimension_type;     /**< Type for dimension values. */
+
+  /**
+   * @brief Constructor.
+   * @details After construction of the class, the zigzag filtration should be given in a streaming like way, i.e.,
+   * call @ref insert_face, @ref remove_face or @ref apply_identity for each step of the filtration in order of
+   * the filtration. The bars of the diagram are retrieved via the given callback method every time
+   * a pair with non-zero length is closed. To retrieve the open/infinite bars, use @ref get_current_infinite_intervals.
```

Comment:
I agree that it should be in the doc of the class. But I like it also at the constructor. So, like last time I just put it in both.

Reply:
Can you explain the logic? I am likely to make the same comment in other code you write, so it would help if I understand the reasoning.
To me, how to use a class is something that should be documented at the level of the class, or even higher (module for instance), while the documentation of a function should give details on how to use this specific function. In some cases, it makes sense for the doc of a function to talk about the follow-up, for instance the documentation of `allocate` should say that the returned value is supposed to be passed to `deallocate` once we are done with it. However, this constructor doesn't have anything specific. If the class had 2 constructors, would you repeat the doc of the class in both?

Reply:
I tend to read the documentation for the methods more than the class documentation. But in this particular case,  the real reason is that it is not intuitive for every user, that the constructor does not take the whole filtration as argument like all other zigzag libraries. So I felt the need to explain how it works at this place. And as I said somewhere else already, redundancy in documentation is rarely a bad thing.